librz/arch/isa/mips/il/mips32.c

```diff
@@ -0,0 +1,1435 @@
+	RzILOpEffect *set_lo = SETG(MIPS32_REG_LO, res_lo);
+
+	return SEQ3(res, set_hi, set_lo);
+}
+
+static RzILOpEffect *mips32_il_neg_fmt(csh *handle, cs_insn *insn) {
+	// Floating Point Negate
+	return NULL;
+}
+
+static RzILOpEffect *mips32_il_nmadd_fmt(csh *handle, cs_insn *insn) {
+	// Floating Point Negative Multiply Add
+	return NULL;
+}
+
+static RzILOpEffect *mips32_il_nmsub_fmt(csh *handle, cs_insn *insn) {
+	// Floating Point Negative Multiply Subtract
+	return NULL;
+}
+
+static RzILOpEffect *mips32_il_nor(csh *handle, cs_insn *insn) {
+	if (REG_IS_ZERO(0)) {
+		return NOP(); // return NOP if target register is $zero
+	}
+
+	const char *rd = REG(0);
+	RzILOpPure *rs = MIPS32_REG(1);
+	RzILOpPure *rt = MIPS32_REG(2);
+
+	return SETG(rd, LOGNOT(LOGOR(rs, rt)));
+}
+
+static RzILOpEffect *mips32_il_or(csh *handle, cs_insn *insn) {
+	if (REG_IS_ZERO(0)) {
+		return NOP(); // return NOP if target register is $zero
+	}
+
+	if (OPCOUNT() < 3) {
+		// move
+		return mips32_il_move(handle, insn);
+	}
+
+	const char *rd = REG(0);
+	if (REG_IS_ZERO(1) && REG_IS_ZERO(2)) {
+		RzILOpPure *zero = SN(MIPS32_GPRLEN, 0);
+		return SETG(rd, zero);
+	} else if (REG_IS_ZERO(1)) {
+		RzILOpPure *rt = MIPS32_REG(2);
+		return SETG(rd, rt);
+	} else if (REG_IS_ZERO(2)) {
+		RzILOpPure *rs = MIPS32_REG(1);
+		return SETG(rd, rs);
+	}
+
+	RzILOpPure *rs = MIPS32_REG(1);
+	RzILOpPure *rt = MIPS32_REG(2);
+
+	return SETG(rd, LOGOR(rs, rt));
+}
+
+static RzILOpEffect *mips32_il_ori(csh *handle, cs_insn *insn) {
+	if (REG_IS_ZERO(0)) {
+		return NOP(); // return NOP if target register is $zero
+	}
+
+	const char *rd = REG(0);
+
+	if (REG_IS_ZERO(1)) {
+		// this also covers when IMM(2) is 0
+		RzILOpPure *imm = MIPS32_IMM(2);
+		return SETG(rd, imm);
+	} else if (IMM(2) == 0) {
+		RzILOpPure *rs = MIPS32_REG(1);
+		return SETG(rd, rs);
+	}
+
+	RzILOpPure *rs = MIPS32_REG(1);
+	RzILOpPure *imm = MIPS32_IMM(2);
+
+	return SETG(rd, LOGOR(rs, imm));
+}
+
+static RzILOpEffect *mips32_il_pll_ps(csh *handle, cs_insn *insn) {
+	// Pair Lower Lower
+	return NULL;
+}
+
+static RzILOpEffect *mips32_il_plu_ps(csh *handle, cs_insn *insn) {
+	// Pair Lower Upper
+	return NULL;
+}
+
+static RzILOpEffect *mips32_il_pul_ps(csh *handle, cs_insn *insn) {
+	// Pair Upper Lower
+	return NULL;
+}
+
+static RzILOpEffect *mips32_il_puu_ps(csh *handle, cs_insn *insn) {
+	// Pair Upper Upper
+	return NULL;
+}
+
+static RzILOpEffect *mips32_il_rdhwr(csh *handle, cs_insn *insn) {
+	// Read Hardware Register
+	return NULL;
+}
+
+static RzILOpEffect *mips32_il_rdpgpr(csh *handle, cs_insn *insn) {
+	// Read GPR from Previous Shadow Set
+	return NULL;
+}
+
+static RzILOpEffect *mips32_il_recip_fmt(csh *handle, cs_insn *insn) {
+	// Reciprocal Approximation
+	return NULL;
+}
+
+static RzILOpEffect *mips32_il_rotr(csh *handle, cs_insn *insn) {
+	// Rotate Word Right
+	if (REG_IS_ZERO(0)) {
+		return NOP(); // return NOP if target register is $zero
```

Comment:
```suggestion
		return NOP();
```

Better describe why `NOP` is executed when it is a `zero_reg`.

Reply:
I don't think this is needed, I could just remove all the comments like that since this applies to any command which tries to write to `$zero`